src/database/schemas.ts

```diff
@@ -220,23 +220,27 @@ export default class Schemas {
    * @param object Not user input
    */
   static async generateSQL(schema: string, object: string, internalType: string): Promise<string> {
-    await JobManager.runSQL<{ SRCDTA: string }>([
-      `CALL QSYS2.GENERATE_SQL( DATABASE_OBJECT_NAME => ?, DATABASE_OBJECT_LIBRARY_NAME => ?, DATABASE_OBJECT_TYPE => ?
-                              , CREATE_OR_REPLACE_OPTION => '1', PRIVILEGES_OPTION => '0'
-                              , DATABASE_SOURCE_FILE_NAME => '*STMF'
-                              , STATEMENT_FORMATTING_OPTION => '0'
-                              , SOURCE_STREAM_FILE => '/tmp/Q_GENSQL_' concat current_user concat '.sql'
-                              , SOURCE_STREAM_FILE_END_OF_LINE => 'LF'
-                              , SOURCE_STREAM_FILE_CCSID => 1208 )`
-    ].join(` `), { parameters: [object, schema, internalType] });
-    const lines = await JobManager.runSQL<{ LINE: string }>(
-      `select LINE
-         from table( QSYS2.IFS_READ( PATH_NAME => '/tmp/Q_GENSQL_' concat current_user concat '.sql' ) )`
-    );
-
-    const generatedStatement = lines.map( elem => elem.LINE ).join(`\n`);
-
-    return generatedStatement;
+    const instance = getInstance();
+    const connection = instance.getConnection();
+
+    const result = await connection.withTempDirectory<string>(async (tempDir) => {
+      const tempFilePath = path.posix.join(tempDir, `generatedSql.sql`);
```

Comment:
@worksofliam Don't we have an API for generating a unique filename? Your code will collide between users, if both have `/tmp` as temp directory.

Reply:
@chrjorgensen I believe the API you're referring to generates a unique name for a file you're going to use over and over again. But since this API, `withTempDirectory`, deletes the directory after, it is more suited here.

Reply:
@worksofliam Ah, didn't know the `withTempDirectory` function... I should, though, since it was created in April last year by @sebjulliand ...! 😊